platform/android/src/style/layers/fill_layer.cpp

```diff
@@ -40,42 +41,134 @@ namespace android {
         return jni::Object<jni::ObjectTag>(*converted);
     }
 
+    jni::Array<jni::jlong> FillLayer::getFillColorTransition(jni::JNIEnv& env) {
+        using namespace mbgl::android::conversion;
+
+        mbgl::style::TransitionOptions options = layer.as<mbgl::style::FillLayer>()->FillLayer::getFillColorTransition();
+        jlong durations[2];
+        durations[0] = options.duration.value_or(mbgl::Duration::zero()).count();
+        durations[1] = options.delay.value_or(mbgl::Duration::zero()).count();
+        auto output = jni::Array<jni::jlong>::New(env, 2);
+        jni::SetArrayRegion(env, *output, 0, 2, durations);
+        return output;
+    }
+
+    void FillLayer::setFillColorTransition(jni::JNIEnv& env, jlong duration, jlong delay) {
+        (void) env;
+        mbgl::style::TransitionOptions options = {mbgl::Duration(duration), mbgl::Duration(delay)};
+        layer.as<mbgl::style::FillLayer>()->FillLayer::setFillColorTransition(options);
+    }
+
     jni::Object<jni::ObjectTag> FillLayer::getFillOutlineColor(jni::JNIEnv& env) {
         using namespace mbgl::android::conversion;
         Result<jni::jobject*> converted = convert<jni::jobject*>(env, layer.as<mbgl::style::FillLayer>()->FillLayer::getFillOutlineColor());
         return jni::Object<jni::ObjectTag>(*converted);
     }
 
+    jni::Array<jni::jlong> FillLayer::getFillOutlineColorTransition(jni::JNIEnv& env) {
+        using namespace mbgl::android::conversion;
+
+        mbgl::style::TransitionOptions options = layer.as<mbgl::style::FillLayer>()->FillLayer::getFillOutlineColorTransition();
+        jlong durations[2];
+        durations[0] = options.duration.value_or(mbgl::Duration::zero()).count();
+        durations[1] = options.delay.value_or(mbgl::Duration::zero()).count();
+        auto output = jni::Array<jni::jlong>::New(env, 2);
+        jni::SetArrayRegion(env, *output, 0, 2, durations);
+        return output;
+    }
+
+    void FillLayer::setFillOutlineColorTransition(jni::JNIEnv& env, jlong duration, jlong delay) {
+        (void) env;
+        mbgl::style::TransitionOptions options = {mbgl::Duration(duration), mbgl::Duration(delay)};
+        layer.as<mbgl::style::FillLayer>()->FillLayer::setFillOutlineColorTransition(options);
+    }
+
     jni::Object<jni::ObjectTag> FillLayer::getFillTranslate(jni::JNIEnv& env) {
         using namespace mbgl::android::conversion;
         Result<jni::jobject*> converted = convert<jni::jobject*>(env, layer.as<mbgl::style::FillLayer>()->FillLayer::getFillTranslate());
         return jni::Object<jni::ObjectTag>(*converted);
     }
 
+    jni::Array<jni::jlong> FillLayer::getFillTranslateTransition(jni::JNIEnv& env) {
+        using namespace mbgl::android::conversion;
+
+        mbgl::style::TransitionOptions options = layer.as<mbgl::style::FillLayer>()->FillLayer::getFillTranslateTransition();
+        jlong durations[2];
+        durations[0] = options.duration.value_or(mbgl::Duration::zero()).count();
```

Comment:
Can we use the actual unit instead of `Duration` so it's clear on first glance what we're talking about?